### PR TITLE
Columnstore cache

### PR DESCRIPTION
**Pull request by tntnatbry**

### CODE REVIEW

**Comment by tntnatbry** [comment]
This PR has a dependency on the changes in the server branch bb-10.5-monty. Please don't merge until the relevant changes from this branch (https://github.com/MariaDB/server/commit/56972cd0a0a0fa2a4b35e5e6b9dbd549595fa8d9) are merged into server 10.5

File: dbcon/mysql/ha_mcs.cpp
Comment (pleblanc1976) [inline dbcon/mysql/ha_mcs.cpp:1348]: any need to check the length of name here?

File: dbcon/mysql/ha_mcs.cpp
Comment (pleblanc1976) [inline dbcon/mysql/ha_mcs.cpp:1360]: does this need to be in its own { } block?

File: dbcon/mysql/ha_mcs.cpp
Comment (pleblanc1976) [inline dbcon/mysql/ha_mcs.cpp:1225]: I don't see any man pages for these, have to assume they're maria fcns and that they're being used correctly

File: dbcon/mysql/ha_mcs.cpp
Comment (pleblanc1976) [inline dbcon/mysql/ha_mcs.cpp:1648]: another block that seems like it doesn't need to be there

**Comment by pleblanc1976** [comment]
After Monty's branch is merged into the server, we'll need to rebase the bb-10.5-cs branch of the server on 10.5.  Our ci/cd system uses bb-10.5-cs.

**Review by pleblanc1976 — DISMISSED** [review]
Adding a technical blocker to prevent accidental merging until the reqs are in place.

File: dbcon/mysql/ha_mcs_execplan.cpp
Comment (drrtuy) [inline dbcon/mysql/ha_mcs_execplan.cpp]: There will be no Infinidb tables anymore.

File: dbcon/mysql/ha_mcs.cpp
Comment (drrtuy) [inline dbcon/mysql/ha_mcs.cpp]: We should align the version with MCS version.

File: dbcon/mysql/ha_mcs.cpp
Comment (drrtuy) [inline dbcon/mysql/ha_mcs.cpp]: If this is a pointer we better use nullptr instead.

File: dbcon/mysql/ha_mcs.cpp
Comment (drrtuy) [reply dbcon/mysql/ha_mcs.cpp:1648]: There is a LEX_CSTRING name used in this block only so I disagree it is totally useless. Moreover the block might be implemented by Monty so he might have his own reasoning behind the explicit block.

File: dbcon/mysql/ha_mcs.cpp
Comment (drrtuy) [reply dbcon/mysql/ha_mcs.cpp:1225]: Yes. It is defined in strings/strxmov.c

File: dbcon/mysql/CMakeLists.txt
Comment (drrtuy) [inline dbcon/mysql/CMakeLists.txt]: What exactly do we use from perfschema?

**Review by drrtuy — APPROVED** [review]
Technically it is OK. 
I would use the same version as MCS plugin has to avoid ambiguities and check wether we really need to link ha_columnstore against perfschema.

**Comment by drrtuy** [comment]
JFYI bb-10.5-cs has been rebased but not pushed yet so stay tuned.

**Comment by drrtuy** [comment]
The rebased bb-10.5-cs has just been pushed so feel free to check if the relevant commits are there.

File: dbcon/mysql/ha_mcs.cpp
Comment (tntnatbry) [reply dbcon/mysql/ha_mcs.cpp:1348]: I think we are safe here not to; this call is made by the server handler mechanism which should be doing this error handling.

File: dbcon/mysql/ha_mcs.cpp
Comment (tntnatbry) [reply dbcon/mysql/ha_mcs.cpp:1360]: I had the same question. I don't see a need to put this in a dedicated code block. I decided to keep it as I am not sure what the original author of the code thought while doing this.

File: dbcon/mysql/ha_mcs_execplan.cpp
Comment (tntnatbry) [reply dbcon/mysql/ha_mcs_execplan.cpp]: Good catch. This is updated.

File: dbcon/mysql/ha_mcs.cpp
Comment (tntnatbry) [reply dbcon/mysql/ha_mcs.cpp]: Ok. I have changed this to use MCS versions here.

File: dbcon/mysql/ha_mcs.cpp
Comment (tntnatbry) [reply dbcon/mysql/ha_mcs.cpp]: Yes this is pointer to a pointer. I have changed it to use a NULL here instead, just to keep the NULL usage consistent with the rest of the source file. Although there are several other places in the code still where we are still using 0.

File: dbcon/mysql/CMakeLists.txt
Comment (tntnatbry) [reply dbcon/mysql/CMakeLists.txt]: I originally had linker errors when trying to link aria symbols, which indirectly depended on symbols from the myisam and perfschema libs. This was when we still had references of ha_maria hton in the code here. Now that we don't explicitly need ha_maria, I found that the linker does not complain anymore. So I have removed this linkage altogether.

**Comment by tntnatbry** [comment]
I have made changes based on the comments on the PR in the commit https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/1254/commits/1a5c6910da9cc33d64044f01439749de65877ea2 and pushed. 
Please review.

**Pull request merged**
